bcgov/biohubbc
# SIMSBIOHUB-70: Hide Upload Button for Observations and Summary Results Once They Have Been Submitted

# Overview
Ensures that files and data that are submitted from SIMS cannot be resubmitted without requesting to do so.

## Links to Jira tickets

 - https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-70

## Description of relevant changes

- Disabled importing Observations and Summary Results if they have already been submitted to BioHub
- Disabled deleting Observations, Summary Results, Reports or Attachment sif they have already been submitted to BioHub
- Disabled the option to _Remove or Resubmit_ and Obseration, Summary Result, Report or Attachment if it hasn't been submitted to BioHub
- Removed full page reload when navigating from a survey back to the parent project when clicking **Back to Project**
- Refactored the _Remove or Resubmit_ dialog and related components

# Testing Procedure
1. Upload an occurrence submission without submitting it. In the context menu, you should:
 - [ ] See the **Delete** button
 - [ ] NOT see the **Remove or Resubmit** button.

2. Upload a summary submission without submitting it. In the context menu, you should:
 - [ ] See the **Delete** button
 - [ ] NOT see the **Remove or Resubmit** button.

3. Upload a survey document without submitting it. In the context menu, you should:
 - [ ] See the **Delete** button
 - [ ] NOT see the **Remove or Resubmit** button.

4. Upload a project document without submitting it. In the context menu, you should:
 - [ ] See the **Delete** button
 - [ ] NOT see the **Remove or Resubmit** button.

5. Submit each of the occurence submission, summary submission, survey and project documents. You shoud:
 - [ ] See that the **Import** button is disabled for both occurrence submissions and summary results submissions

6. For the context menu in each of the occurence submission, summary submission, survey and project documents, you should:

 - [ ] NOT see the **Delete** button
 - [ ] See the **Remove or Resubmit** button.
